presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlTypeMapping.java

```diff
@@ -556,6 +556,47 @@ public void testTimestamp(boolean legacyTimestamp, boolean insertWithPresto)
         }
     }
 
+    @Test(dataProvider = "testTimestampDataProvider")
+    public void testArrayTimestamp(boolean legacyTimestamp, boolean insertWithPresto)
+    {
+        DataType<List<LocalDateTime>> dataType;
+        DataSetup dataSetup;
+        if (insertWithPresto) {
+            dataType = arrayDataType(timestampDataType());
+            dataSetup = prestoCreateAsSelect("test_array_timestamp");
+        }
+        else {
+            dataType = arrayDataType(timestampDataType(), "timestamp[]");
+            dataSetup = postgresCreateAndInsert("tpch.test_array_timestamp");
+        }
+        // using two non-JVM zones so that we don't need to worry what Postgres system zone is
+        for (ZoneId sessionZone : ImmutableList.of(ZoneOffset.UTC, jvmZone, vilnius, kathmandu, ZoneId.of(TestingSession.DEFAULT_TIME_ZONE_KEY.getId()))) {
+            DataTypeTest tests = DataTypeTest.create()
+                    .addRoundTrip(dataType, asList(beforeEpoch))
+                    .addRoundTrip(dataType, asList(afterEpoch))
+                    .addRoundTrip(dataType, asList(timeDoubledInJvmZone))
+                    .addRoundTrip(dataType, asList(timeDoubledInVilnius));
+
+            if (!insertWithPresto) {
+                // when writing, Postgres JDBC driver converts LocalDateTime to string representing date-time in JVM zone
+                // TODO upgrade driver or find a different way to write timestamp values
+                addTimestampTestIfSupported(tests, legacyTimestamp, sessionZone, epoch); // epoch also is a gap in JVM zone
+                addTimestampTestIfSupported(tests, legacyTimestamp, sessionZone, timeGapInJvmZone1);
+                addTimestampTestIfSupported(tests, legacyTimestamp, sessionZone, timeGapInJvmZone2);
```

Comment:
`addTimestampTestIfSupported` adds a test case with `timestamp` type, not `array(timestamp)`.

(btw this should currently fail, because read path for `timestamp[]` uses `j.sql.Timestamp` and cannot retrieve date/time which is a GAP in JVM zone)

Reply:
Fixed